tests/subsys/bluetooth/mesh/light_ctrl/src/main.c

```diff
@@ -327,28 +327,16 @@ expected_statemachine_cond(bool enabled,
 		      light_ctrl_srv.flags);
 }
 
-static struct sensor_value float_to_sensor_value(float fvalue)
-{
-	return (struct sensor_value) {
-		       .val1 = (int32_t)(fvalue),
-		       .val2 = ((int64_t)((fvalue) * 1000000.0f)) % 1000000L
-	};
-}
-
-static float sensor_to_float(struct sensor_value *val)
-{
-	return val->val1 + val->val2 / 1000000.0f;
-}
-
 static void send_amb_lux_level(float amb)
```

Comment:
Perhaps not strictly needed, but I think it would be better if this took a millis integer argument. Since this takes float now but the encoded format encodes fixed-point, we can potentially run into accuracy problems. (In the case where the source value is a float the result of this function is by definition lossy, but then the caller can convert float to int)

Reply:
I think this use case is OK since the `amb` is a sensor value read as float in raw form and then used for regulator eventually. And it is a valid use case actually IMO, and it is acceptable. Besides `bt_mesh_light_ctrl_reg`'s measured, target and prev_target values are all float, conversion is inevitable.

Reply:
`amb`, when converted to sensor value, is not "float in raw form". The raw form is a fixed point, integer-based value. But I'm not going to spend time arguing about this, as long as the test passes.